extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetMetadataFileWriterImpl.java

```diff
@@ -0,0 +1,177 @@
+package io.deephaven.parquet.table;
+
+import io.deephaven.UncheckedDeephavenException;
+import io.deephaven.parquet.base.ParquetFileReader;
+import io.deephaven.parquet.base.ParquetFileWriter;
+import io.deephaven.parquet.base.ParquetMetadataFileWriter;
+import io.deephaven.parquet.base.PositionedBufferedOutputStream;
+import io.deephaven.parquet.table.metadata.ColumnTypeInfo;
+import io.deephaven.parquet.table.metadata.TableInfo;
+import io.deephaven.util.channel.SeekableChannelsProvider;
+import io.deephaven.util.channel.SeekableChannelsProviderLoader;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.hadoop.metadata.FileMetaData;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.MessageType;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static io.deephaven.util.channel.SeekableChannelsProvider.convertToURI;
+
+/**
+ * Used to generate a combined _metadata and _common_metadata file for provided Parquet files.
+ */
+public final class ParquetMetadataFileWriterImpl implements ParquetMetadataFileWriter {
+
+    /**
+     * A class to hold the parquet file and its metadata.
+     */
+    private static class ParquetFileMetadata {
+        final File file;
+        final ParquetMetadata metadata;
+
+        ParquetFileMetadata(final File file, final ParquetMetadata metadata) {
+            this.file = file;
+            this.metadata = metadata;
+        }
+    }
+
+    /**
+     * A class to accumulate metadata from multiple parquet files.
+     */
+    private static final class AccumulatedMetadata {
```

Comment:
There are a lot of variables with names related to metadata here, let me know if you want me to rename any of these.